ethcore/src/verification/verification.rs

```diff
@@ -264,13 +264,21 @@ pub fn verify_header_params(header: &Header, engine: &dyn Engine, is_full: bool,
 	if header.gas_used() > header.gas_limit() {
 		return Err(From::from(BlockError::TooMuchGasUsed(OutOfBounds { max: Some(*header.gas_limit()), min: None, found: *header.gas_used() })));
 	}
-	let min_gas_limit = engine.params().min_gas_limit;
-	if header.gas_limit() < &min_gas_limit {
-		return Err(From::from(BlockError::InvalidGasLimit(OutOfBounds { min: Some(min_gas_limit), max: None, found: *header.gas_limit() })));
-	}
-	if let Some(limit) = engine.maximum_gas_limit() {
-		if header.gas_limit() > &limit {
-			return Err(From::from(BlockError::InvalidGasLimit(OutOfBounds { min: None, max: Some(limit), found: *header.gas_limit() })));
+	if let Some(gas_limit) = engine.gas_limit_override(header) {
+		if *header.gas_limit() != gas_limit {
+			return Err(From::from(BlockError::InvalidGasLimit(
+				OutOfBounds { min: Some(gas_limit), max: Some(gas_limit), found: *header.gas_limit() }
+			)));
+		}
+	} else {
+		let min_gas_limit = engine.params().min_gas_limit;
```

Comment:
Seems like there should be an `engine.min_gas_limit()` method (or perhaps `engine.gas_limits() -> (u64, u64)`?).

Reply:
I'll add a method although it's redundant because the field `min_gas_limit` is public.

Reply:
`min_gas_limit` being public is… not that great! There's probably some historical reason for that.

The point I was trying to make was this: I'm a bit hesitant about this PR because it adds code that is useful only to Aura. Ideally we should architect things such that engine-specific behaviour stays in that engine's code. One idea I had was to have a single `gas_limits()` method on `Engine` which would let the Aura implementation to run the logic it needs without "polluting" other engines.

Reply:
This can be done, I think.
@varasev, is it OK to move `blockGasLimitContract` from the general engine parameter set to the AuRa parameter set? I would also recommend renaming it as `blockGasLimitContracts` since we are changing it to support only a map.

Reply:
Yes, it's OK to move, but don't rename, please, because the map still assumes that there is only one contract at a time.

If you move this parameter, please also move it in our https://github.com/poanetwork/parity-ethereum/tree/aura-pos branch.

Reply:
@varasev OK, I'll copy that in `aura-pos`. Are you sure about the name? Wouldn't plural look better alongside `blockRewardContractTransitions`?

Reply:
Then that should be named `blockGasLimitContractTransitions`

Reply:
Makes sense.

Reply:
Done in https://github.com/poanetwork/parity-ethereum/pull/189.